python/ray/util/sgd/v2/trainer.py

```diff
@@ -244,8 +264,75 @@ def to_tune_trainable(
         Returns:
             A Trainable that can directly be passed into ``tune.run()``.
         """
+        if not TUNE_INSTALLED:
+            raise ValueError("Tune is not installed. Please install ray["
+                             "tune] to use the Tune integration.")
 
-        def trainable_func(config: Dict[str, Any]) -> T:
-            pass
+        return _create_tune_trainable(train_func, self._backend,
+                                      self._num_workers, self._use_gpu,
+                                      self._resources_per_worker)
 
-        raise NotImplementedError
+
+class _TuneTrainer(Trainer):
+    def start_training(
+            self,
+            train_func: Union[Callable[[], T], Callable[[Dict[str, Any]], T]],
+            config: Optional[Dict[str, Any]] = None) -> None:
+
+        train_func = self._get_train_func(train_func, config)
+        self._executor.start_training(train_func)
+
+    def fetch_next_report(self) -> Optional[List[Dict]]:
+        return self._executor.fetch_next_result()
+
+    def finish_training(self) -> List[T]:
+        return self._executor.finish_training()
```

Comment:
btw, why do inheritance here? 

This seems a bit like a misuse of inheritance, since it doesn't actually seem like you're using any of the interface you're inheriting.

https://softwareengineering.stackexchange.com/questions/12439/code-smell-inheritance-abuse

Reply:
Yep I removed this and am using the `run_iterator` API instead.